### ISSUE TITLE
Wrong Alignment of Datatable Column headers in combination with ColumnToggler 

### ISSUE THREAD

**Issue by ademcaliskan** [primefaces/primefaces#187]
PF-Version: 5.1
Mojarra-Version: Mojarra 2.1.28-jbossorg-2
App-Server: JBoss EAP 6.3

When I use Datatable in combination with ColumToggler in Header, like described in the showcase, then the column header are aligned not on top the the corresponding column.

See attached screenshot.

![image](https://cloud.githubusercontent.com/assets/2758366/7366363/0c759f04-ed98-11e4-9582-6361e50e59e3.png)

**Comment by cagataycivici** [comment]
More information please such as code. And try with 5.2 as well.

**Comment by ademcaliskan** [comment]
Seems to be an issue, if using scrollable table.

Modified code from showcase.

``` xhtml
<?xml version='1.0' encoding='UTF-8' ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:p="http://primefaces.org/ui"
      xmlns:f="http://java.sun.com/jsf/core">
<f:view>
    <h:head>
        <title>JUIS Rule Management</title>
        <link rel="shortcut icon" type="image/x-icon" href="#{request.contextPath}/resources/images/favicon.ico"/>
        <h:outputStylesheet library="css" name="pf_customized.css"/>
        <h:outputStylesheet library="css" name="main.css"/>
    </h:head>
    <h:body>
        <h:outputLabel value="PF-Datatable"/>
        <p:dataTable id="cars" var="car" value="#{dtBasicView.cars}"
                scrollable="true"
                scrollHeight="100">
            <f:facet name="header">
                List of Cars

                <p:commandButton id="toggler" type="button" value="Columns" style="float:right"
                                 icon="ui-icon-calculator"/>
                <p:columnToggler datasource="cars" trigger="toggler"/>
            </f:facet>

            <p:column headerText="Id">
                <h:outputText value="#{car.id}"/>
            </p:column>

            <p:column headerText="Year">
                <h:outputText value="#{car.year}"/>
            </p:column>

            <p:column headerText="Brand">
                <h:outputText value="#{car.brand}"/>
            </p:column>

            <p:column headerText="Color">
                <h:outputText value="#{car.color}"/>
            </p:column>

            <p:column headerText="Price">
                <h:outputText value="#{car.price}">
                    <f:convertNumber type="currency" currencySymbol="$"/>
                </h:outputText>
            </p:column>
        </p:dataTable>
    </h:body>
</f:view>
</html>
```

**Comment by ademcaliskan** [comment]
Another point is when using paginator.

![image](https://cloud.githubusercontent.com/assets/2758366/7370392/1829be58-edba-11e4-80dd-e4969c0f7761.png)

``` xhtml
<?xml version='1.0' encoding='UTF-8' ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:p="http://primefaces.org/ui"
      xmlns:f="http://java.sun.com/jsf/core">
<f:view>
    <h:head>
        <title>JUIS Rule Management</title>
        <link rel="shortcut icon" type="image/x-icon" href="#{request.contextPath}/resources/images/favicon.ico"/>
        <h:outputStylesheet library="css" name="pf_customized.css"/>
        <h:outputStylesheet library="css" name="main.css"/>
    </h:head>
    <h:body>
        <h:outputLabel value="PF-Datatable"/>
        <p:dataTable id="cars" var="car" value="#{dtBasicView.cars}"
                     paginator="true"
                     rows="3"
                >
            <f:facet name="header">
                List of Cars

                <p:commandButton id="toggler" type="button" value="Columns" style="float:right"
                                 icon="ui-icon-calculator"/>
                <p:columnToggler datasource="cars" trigger="toggler"/>
            </f:facet>

            <p:column headerText="Id">
                <h:outputText value="#{car.id}"/>
            </p:column>

            <p:column headerText="Year">
                <h:outputText value="#{car.year}"/>
            </p:column>

            <p:column headerText="Brand">
                <h:outputText value="#{car.brand}"/>
            </p:column>

            <p:column headerText="Color">
                <h:outputText value="#{car.color}"/>
            </p:column>

            <p:column headerText="Price">
                <h:outputText value="#{car.price}">
                    <f:convertNumber type="currency" currencySymbol="$"/>
                </h:outputText>
            </p:column>
        </p:dataTable>
    </h:body>
</f:view>
</html>
```

To reproduce use given xhtml-document:
- show table
- hide one col
- and go to another pager over paginator

**Comment by cagataycivici** [comment]
Please do a search before creating new issues, this one is a duplicate of #41

**Comment by andrespontt** [comment]
I think this should be the expected behavior.

http://plnkr.co/edit/xQzMy1TVd8zmpOYZA2tf?p=preview

**Comment by cagataycivici** [comment]
For scrolling;
#41

For others, you need to implement something like;
http://blog.primefaces.org/?p=3341

**Comment by ademcaliskan** [comment]
Seems straight forward. Will give it a try, but I am confident that it will work. 

Thx

Adem

**Comment by skyfall92** [comment]
Has this bug been corrected in Primefaces 5.0?
